lib/cocoapods/user_interface.rb

```diff
@@ -372,7 +374,11 @@ def gets
       # return [void]
       #
       def warn(message, actions = [], verbose_only = false)
-        warnings << { :message => message, :actions => actions, :verbose_only => verbose_only }
+        if treat_warnings_as_errors
+          raise Informative, message
```

Comment:
this might not be the right place to add this. This is meant to add all warnings and display them in the end.

I would push that to the `installer.rb` instead.

Reply:
The problem with that approach is that the warnings which may oe encountered during installation aren't just in installer.rb. The warning in the originating issue, for example, is in [Xcodeproj/lib/xcodeproj/project/object.rb](https://github.com/CocoaPods/Xcodeproj/blob/master/lib/xcodeproj/project/object.rb).

It would be possible to check afterward in installer.rb whether any warnings have triggered, but that would violate the expected behavior that errors cause an immediate halt in execution.